aspnet/migration/http-modules.rst

```diff
@@ -1,8 +1,576 @@
+	:language: c#
+	:linenos:
+	:emphasize-lines: 6, 8, 24, 31
+
+As shown in the :doc:`../fundamentals/middleware` page,
+an ASP.NET 5 middleware is simply a class that exposes an ``Invoke`` method taking an ``HttpContext`` and returning a ``Task``. Your new middleware will look like this:
+
+.. _http-modules-usemiddleware:
+
+.. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/MyMiddleware.cs
+	:language: c#
+	:linenos:
+	:emphasize-lines: 9, 13, 20, 24, 28, 30, 32
+
+The middleware template used here is the same as that shown in the section on 
+:ref:`writing middleware <middleware-writing-middleware>`.
+
+The `MyMiddlewareExtensions` helper class makes it easier to configure your middleware in your ``Startup`` class. 
+The ``UseMyMiddleware`` method adds your middleware class to the request pipeline. Services required by the middleware get injected in the middleware's constructor.
+
+Maybe your module sometimes terminates the request, for example if the user is not authorized:
+
+.. literalinclude:: http-modules/sample/Asp.Net4/Asp.Net4/Modules/MyTerminatingModule.cs
+	:language: c#
+	:linenos:
+	:emphasize-lines: 9-13
+	:lines: 18-31
+	:dedent: 8
+
+A middleware handles this by simply not calling ``Invoke`` on the next middleware in the pipeline. Keep in mind that this does not 
+fully terminate the request, because previous middlewares will still be invoked when the response makes it way back through the pipeline
+to the browser.
+
+.. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/MyTerminatingMiddleware.cs
+	:language: c#
+	:linenos:
+	:emphasize-lines: 7, 8
+	:lines: 16-26
+	:dedent: 8
+
+When you migrate your module's functionality to your new middleware, you may find that your code doesn't compile because the ``HttpContext`` class 
+has changed greatly in ASP.NET 5. `Later on <#migrating-to-the-new-httpcontext>`_, you'll see how to migrate to the new ASP.NET 5 HttpContext.
+
+Migrating module insertion into the request pipeline
+-----------------------------------------------------
+HTTP modules are typically added to the request pipeline using web.config:
+
+.. literalinclude:: http-modules/sample/Asp.Net4/Asp.Net4/Web.config
+	:language: xml
+	:linenos:
+	:emphasize-lines: 6
+	:lines: 1-3, 29-30, 33, 40, 48, 105
+
+Convert this by `adding your new middleware <../fundamentals/middleware.html#creating-a-middleware-pipeline-with-iapplicationbuilder>`_
+to the request pipeline in your ``Startup`` class:
+
+.. literalinclude:: http-modules/sample/Asp.Net5/src/Asp.Net5/Startup.cs
+	:language: c#
+	:linenos:
+	:lines: 1-2, 12-15, 46-48, 54-58, 105, 109, 110
+	:emphasize-lines: 12
+
+The exact spot in the pipeline where you insert your new middleware depends on the event 
+that it handled as a module (``BeginRequest``, ``EndRequest``, etc.) and its order in your list of modules in your web.config.
+
+However, as you saw there is no more application life cycle in ASP.NET 5
```

Comment:
As previously stated, there is no more application life cycle in ASP.NET 5 and the order in which responses are processed by middleware differs from the order used by modules.